### DISCUSSION TITLE
positive part of a vector

### DISCUSSION THREAD

**Discussion by ksurana21** [yalmip/YALMIP#944] [Optimization modelling] [answered]
Hello,

I wanted to ask if the [subplus function of Matlab](https://uk.mathworks.com/help/curvefit/subplus.html) is supported by YALMIP.

I am asking this because I want to add constraints that look like this:
z == (max{x-y,0})^2 
where x, y, and z are all decision variables. 

I have tried using the subplus function with YALMIP in the following manner:
z == subplus(x-y).^2
It seems to work without errors but the solutions I am getting seem to be incorrect. Is it possible that YALMIP does not recognize this function? Is there any other way of including such constraints?

Thank you for the help!

**Comment by johanlofberg** [comment, marked as answer]
subplus is not overloaded, why not simply write max(x-y,0)^2?

Note though that you will get a nonconvex quadratic model with binary variables. If you use the max^2 operator in a convex way, you are better of to do the modelling manually, as squaring a convex function does not preserve convexity and thus forces YALMIP to go to a MLP model

Hence, if you have the max^2 operator entering in a convex way, you should introduce an epigraph variable t and use t^2 with the constraints t>=0, t>=x-y

There is also https://yalmip.github.io/command/hinge/, although I think it requires you to install the develop version
